homeassistant/components/swidget/config_flow.py

```diff
@@ -0,0 +1,211 @@
+            host=discovered_ip,
+            mac=discovered_mac,
+            device_type=device_type,
+            insert_type=insert_type,
+            friendly_name=friendly_name,
+        )
+
+    async def async_step_integration_discovery(
+        self, discovery_info: DiscoveryInfoType
+    ) -> ConfigFlowResult:
+        """Handle integration discovery."""
+        return await self._async_handle_discovery(
+            discovery_info[CONF_HOST], discovery_info[CONF_MAC]
+        )
+
+    async def _async_handle_discovery(
+        self,
+        host: str,
+        mac: str,
+        device_type: str = "Unknown",
+        insert_type: str = "Unknown",
+        friendly_name: str = "Unknown Swidget Device",
+    ) -> ConfigFlowResult:
+        """Handle any discovery."""
+        await self.async_set_unique_id(format_mac(mac))
+        self._abort_if_unique_id_configured()
+
+        self.context[CONF_HOST] = host
+        for progress in self._async_in_progress():
+            if progress.get("context", {}).get(CONF_HOST) == host:
+                return self.async_abort(reason="already_in_progress")
+        self._discovered_device = SwidgetDiscoveredDevice(
+            mac, host, device_type, insert_type, friendly_name
+        )
+        return await self.async_step_discovery_confirm()
+
+    async def async_step_discovery_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm discovery."""
+        assert self._discovered_device is not None
+        if user_input is not None:
+            user_input[CONF_HOST] = self._discovered_device.host
+            info = await validate_input(self.hass, user_input)
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        self._set_confirm_only()
+        placeholders = {
+            "name": self._discovered_device.friendly_name,
+            "host": self._discovered_device.host,
+        }
+        self.context["title_placeholders"] = placeholders
+        return self.async_show_form(
+            step_id="discovery_confirm",
+            data_schema=vol.Schema({vol.Required("password"): str}),
+            description_placeholders=placeholders,
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            if not (user_input[CONF_HOST]):
+                return await self.async_step_pick_device()
+            try:
+                info = await validate_input(self.hass, user_input)
+                mac_address = info["mac_address"]
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unable to validate Swidget device credentials")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(format_mac(mac_address))
```

Comment:
```suggestion
                await self.async_set_unique_id(format_mac(mac_address), raise_in_progress=False)
```
This will make sure the user flow is continued and any other (discovery) flow is aborted

Reply:
Fixed I believe

Reply:
You've not set the `raise_in_progress` flag to False though.

If you don't set it to False, `raise_in_progress` aborts if there are other config flows in progress for the same MAC address. If you do set it to False we instead abort the *other* flow with same unique id. We want the user flow to have priority, so we should abort the other flows.